src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -965,18 +965,13 @@ private object ExecuteClusterTransaction(ClusterOperationContext context, Blitta
                 var errors = clusterTransaction.ExecuteCompareExchangeCommands(rawRecord.GetClusterTransactionId(), context, index, compareExchangeItems);
                 if (errors == null)
                 {
-                    DatabasesLandlord.ClusterDatabaseChangeType notify;
                     var clusterTransactionResult = new ClusterTransactionResult();
                     if (clusterTransaction.HasDocumentsInTransaction)
                     {
                         clusterTransactionResult.GeneratedResult = clusterTransaction.SaveCommandsBatch(context, rawRecord, index);
-                        notify = DatabasesLandlord.ClusterDatabaseChangeType.PendingClusterTransactions;
-                    }
-                    else
-                    {
-                        notify = DatabasesLandlord.ClusterDatabaseChangeType.ClusterTransactionCompleted;
                     }
 
+                    var notify = DatabasesLandlord.ClusterDatabaseChangeType.PendingClusterTransactions;
```

Comment:
We changed the implementation.
A cluster-wide transaction without a database command will not be written to the `ClusterStateMachine.TransactionCommands` table.
Instead, if we don't find any new commands we notify according to the last compare exchange index